### DISCUSSION TITLE
Ability to configure Jellyfin to use an external SQL Database

### DISCUSSION THREAD

**Discussion by ruphuselderbeer** [jellyfin/jellyfin#12844] [General]
I think it would be a nice option for Jellyfin to be able to use an external SQL database such as MySQL, Postgres SQL, or others. For me at least, I could then more easily reference the same place for user auth across multiple apps/services I run internally. It would also allow me to make more automated database backups.

**Comment by oddstr13** [comment]
Existing feature request over here; https://features.jellyfin.org/posts/315/mysql-server-back-end

TLDR; planned, but the road is long.

**Comment by bkowens** [comment]
I've been hoping this feature would get introduced but I decided to let Claude take a crack at it.  I've been testing a fork of the repo with some enhancements and it handles mysql quite nicely!:
https://github.com/bkowens/JellyFinhanced

**Comment by JPVenson** [comment]
External database support has been added with 10.11 as an __Experimental__ feature. Plugins like https://github.com/JPVenson/Jellyfin.Pgsql can now provide new EFCore compatiable database adapters.

**Comment by canepan** [comment]
Following up on @JPVenson's note about 10.11's plugin-based database providers: there's now a MySQL/MariaDB provider plugin for **stock** Jellyfin (no fork needed):

https://github.com/canepan/jellyfin-plugin-mysql

- Works with the official `jellyfin/jellyfin:10.11.11` image — install the plugin, edit `database.xml`, restart
- Plugin catalog repo available: https://canepan.github.io/jellyfin-plugin-mysql/manifest.json
- Migration from an existing SQLite instance is possible using Jellyfin's built-in backup/restore — I moved my real library (~49k items, 4 users, watch states and resume positions included) onto a MariaDB Galera cluster this way; the
procedure and its one gotcha are in the README

**Status: experimental**. It's version-pinned to 10.11.11 (the migrations are generated against that exact entity model), tested so far with one real-world library on MariaDB 11.4. Modeled on Jellyfin.Pgsql, GPL-3.0. Issues and test reports
very welcome — especially from different data shapes and MySQL 8.x.
